recipes/protobuf/all/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+set(CMAKE_VERBOSE_MAKEFILE ON)
+
+# Download and build .proto
+file(READ ${CMAKE_BINARY_DIR}/conaninfo.txt CONAN_INFO)
+string(REGEX MATCH "([0-9]+\\.[0-9]+\\.[0-9]+)" PACKAGE_VERSION ${CONAN_INFO})
+if(NOT DEFINED PACKAGE_VERSION)
+  message(FATAL_ERROR "Protobuf version not found")
+endif()
+string(TOLOWER ${CONAN_SETTINGS_OS} OS)
+set(FILENAME "${CMAKE_CURRENT_SOURCE_DIR}/protoc.zip")
+set(EXT "")
+if(OS STREQUAL "windows")
+  if(CONAN_SETTINGS_ARCH STREQUAL "x86_64")
+    set(ARCH "64")
+  elseif(CONAN_SETTINGS_ARCH STREQUAL "x86_32")
+    set(ARCH "32")
+  endif()  
+  set(EXT ".exe")
+  set(OS_ARCH "win${ARCH}")
+elseif(OS STREQUAL "macos")
+  set(OS_ARCH "osx-${CONAN_SETTINGS_ARCH}")
+else()
+  set(OS_ARCH "${OS}-${CONAN_SETTINGS_ARCH}")
+endif(OS STREQ "Windows")
+
+set(URL "https://github.com/protocolbuffers/protobuf/releases/download/v${PACKAGE_VERSION}/protoc-${PACKAGE_VERSION}-${OS_ARCH}.zip")
```

Comment:
Why do you need protoc from author's repo? The correct way making protoc package available as dependency

Reply:
I agree but unfortunately, the build of protoc was not considered according to the original patch file. Thus, I just used the most convenient way and grabbed the original protoc compiler binaries.

Reply:
@uilianries Is this approach a blocking one? The libprotoc version is not available in every package and furthermore, I do not want to build my own protoc implementation which es error prone too.

Reply:
Can you avoid protoc here? You can build the proto files and export the .cc file here. Otherwise, you will need to provide the protoc package. However, since 1.24 the cross-building support is better and now we have a new effort putting protobuf and protoc together at same package: https://github.com/conan-io/conan-center-index/pull/1179

Reply:
@uilianries At the moment I cannot not avoid protoc because I have to compile the address.proto file in order to generate the versions corresponding .cc and .h file. At the moment, this works quite well. But I would wait until #1179 is ready for the final merge. How can I address the requirement in the test_package the right way?

Reply:
Okay, let's keep your version for now. It's working and cross-building won't run.